### ISSUE TITLE
[BUG] -DCMAKE_BUILD_TYPE=Debug Fortran_FLAGS missing comma

### ISSUE THREAD

**Issue by TerribleNews** [geoschem/GCHP#35]
When I do cmake with the latest main branch, I get a compile error which I traced back to `src/MAPL/MAPL_cfio_r4/CMakeFiles/MAPL_cfio_r4.dir/flags.make`.  The Fortran_FLAGS variable includes `-check bounds uninit` but for ifort these options should be comma separated, i.e. `-check bounds,uninit`

**Comment by lizziel** [comment]
Thanks for reporting this. I'll put a fix in for our fork of MAPL used in GCHPctm. Would you like to also report this issue at https://github.com/GEOS-ESM/MAPL? Otherwise I can do that.

**Comment by lizziel** [comment]
Hmm, I think I misunderstood your post. By src/ do you mean in the build directory rather than the source code?

**Comment by TerribleNews** [comment]
Yes, sorry. When I filed the bug I hadn't noticed that I was in the build directory. I'm not great at cmake yet. I've done a bit more poking around but I can't figure out for the life of me where it comes from. It's in all the flags.make files that are generated, which is a lot.

**Comment by lizziel** [comment]
No worries, I'll try to reproduce and see if I can track it down.

**Comment by lizziel** [comment]
Following up, I am indeed running into an issue when building with ifort (ifort18 in my case) with debug flags on. The error I get is:
```
[ 15%] Building Fortran object src/MAPL/MAPL_cfio_r4/CMakeFiles/MAPL_cfio_r4.dir/ESMF_CFIOBaseMod.f.o
ifort: error #10236: File not found:  'uninit'
make[2]: *** [src/MAPL/MAPL_cfio_r4/CMakeFiles/MAPL_cfio_r4.dir/ESMF_CFIOBaseMod.f.o] Error 1
make[2]: *** Deleting file `src/MAPL/MAPL_cfio_r4/CMakeFiles/MAPL_cfio_r4.dir/ESMF_CFIOBaseMod.f.o'
make[1]: *** [src/MAPL/MAPL_cfio_r4/CMakeFiles/MAPL_cfio_r4.dir/all] Error 2
make: *** [all] Error 2
```
I assume this is the same issue you ran into and am looking into it.

**Comment by lizziel** [comment]
This error is resulting from the inclusion of `-check uninit` in the default debug flags used in GEOS libraries when building with Intel. That flag is set in the NASA GMAO `ESMA_cmake `repository, in file `Intel.cmake`. The full set of GEOS debug flags defined there are applied when in debug mode for GCHP.

Bypass this issue by going to your main source code directory, change into subdirectory `ESMA_cmake`, and open file `Intel.cmake`. Go to the section that says 

> GEOS Debug

Then change this line:

> set (GEOS_Fortran_Debug_Flags "${DEBINFO} ${FOPT0} ${FTZ} ${ALIGN_ALL} ${NO_ALIAS} -debug -nolib-inline -fno-inline-functions -assume protect_parens,minus0 -prec-div -prec-sqrt -check bounds -check uninit -fp-stack-check -warn unused -init=snan,arrays -save-temps")

to this (`-check uninit `removed):

> set (GEOS_Fortran_Debug_Flags "${DEBINFO} ${FOPT0} ${FTZ} ${ALIGN_ALL} ${NO_ALIAS} -debug -nolib-inline -fno-inline-functions -assume protect_parens,minus0 -prec-div -prec-sqrt -check bounds -fp-stack-check -warn unused -init=snan,arrays -save-temps")

**Comment by TerribleNews** [comment]
A perfect, thanks! Rather than removing uninit, I recommend changing `-check bounds -check uninit` to `-check bounds,uninit`.  I just tried that and it works and it maintains the original intent, I think.

**Comment by lizziel** [comment]
Excellent, thanks!

**Comment by lizziel** [comment]
This is fixed in 13.0.0. Running GCHPctm with debug flags on is now possible with both Intel and GNU fortran compilers.
